CenterForOpenScience/osf.io
# [PLAT-218][Needs WB PR] Use WB download action callbacks to tally preprint views

## Purpose

We want to use addons.base.views.create_waterbutler_log to tally preprint views, this allows us to do that.

**Remember this is for views, not downloads, and only views from non-contributors.**

## Changes

- remove update_analitics from osfstorage_download function
- Change log behavior to increment download_counts during waterbutler callback.
- Add tests.

## QA Notes

You will have to run the migrations and check via the shell as there's nothing user facing in this PR. Also remember contributor downloads don't count in the total.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/PLAT-218
Ticket with more discussion info:
https://openscience.atlassian.net/browse/PLAT-388
Waterbutler PR:
https://github.com/CenterForOpenScience/waterbutler/pull/324
